HVZ.Web/Shared/User Settings/PasswordChange.razor

```diff
@@ -0,0 +1,165 @@
+                </div>
+                <div class="modal-footer">
+                    <button type="submit" id="submit-btn" class="btn btn-primary">Submit</button>
+                    <a class="btn btn-secondary" data-bs-toggle="modal" data-bs-target="#change-password">Cancel</a>
+                </div>
+            </div>
+        </div>
+    </div>
+</EditForm>
+
+
+@code {
+
+    public string? CurrentPassword { get; set; }
+    List<string> currentPasswordErrors = new List<string>();
+    string? CurrentPasswordInvalidClass => currentPasswordErrors.Count > 0 ? "is-invalid" : null;
+
+    [PasswordValidation]
+    public string? NewPassword { get; set; }
+    List<string> newPasswordErrors = new List<string>();
+    string? NewPasswordInvalidClass => newPasswordErrors.Count > 0 ? "is-invalid" : null;
+
+    public string? ConfirmNewPassword { get; set; }
+    List<string> confirmPasswordErrors = new List<string>();
+    string? ConfirmPasswordInvalidClass => confirmPasswordErrors.Count > 0 ? "is-invalid" : null;
+
+    List<string> changePasswordErrors = new List<string>();
+
+    bool changeSuccess = false;
+
+    ApplicationUser appUser = null!;
+
+    void ClearErrors()
+    {
+        currentPasswordErrors = new List<string>();
+        newPasswordErrors = new List<string>();
+        confirmPasswordErrors = new List<string>();
+        changePasswordErrors = new List<string>();
+        changeSuccess = false;
+    }
+
+    void ClearModal()
+    {
+        ClearErrors();
+
+        CurrentPassword = null;
+        NewPassword = null;
+        ConfirmNewPassword = null;
+    }
+
+    async Task HandleSubmit(EditContext editContext)
+    {
+        ClearErrors();
+
+        bool errors = false;
+
+        if (CurrentPassword is null || CurrentPassword == string.Empty) currentPasswordErrors.Add("Required");
+        else if (!(await UserManager.CheckPasswordAsync(appUser, CurrentPassword))) currentPasswordErrors.Add("Incorrect password");
+
+        if (NewPassword is null || NewPassword == string.Empty) newPasswordErrors.Add("Required");
+        if (!editContext.Validate())
+        {
+            foreach (var message in editContext.GetValidationMessages())
+            {
+                newPasswordErrors.Add(message);
+            }
+        }
+
+        if (ConfirmNewPassword is null || ConfirmNewPassword == string.Empty) confirmPasswordErrors.Add("Required");
+        if (ConfirmNewPassword != NewPassword) confirmPasswordErrors.Add("Passwords must match");
+
+        errors = (currentPasswordErrors.Count + newPasswordErrors.Count + confirmPasswordErrors.Count) > 0;
+
+        if (errors) return;
+
+        var result = await UserManager.ChangePasswordAsync(appUser, CurrentPassword!, NewPassword!);
+
+        if (result.Succeeded)
+        {
+            // Display success notification
+            // Email the user
+            // Close the modal
+            changeSuccess = true;
+            return;
+        }
```

Comment:
are these comments TODO?

Reply:
they're eventual TODOs - I don't have the time to write the email templates and the JS

Reply:
create an issue for tracking